lib/thrift/generator/server/binary_framed.ex

```diff
@@ -95,7 +95,7 @@ defmodule Thrift.Generator.Server.BinaryFramed do
       Thrift.TApplicationException ->
         quote do
           unhandled ->
-            Logger.error("Unhandled exception: #{Exception.message(unhandled)}")
+            Logger.error("Exception not defined in thrift spec was thrown: #{Exception.message(unhandled)}")
```

Comment:
`rescue` does not rescue all exceptions, only error exceptions. To rescue all exceptions you require `catch`. I think the current behaviour hides too much information. I think this should be written as:

```elixir
catch
  kind, reason ->
    stack = System.stacktrace()
end
```
The TApplicationException could the use `Exception.format(kind, reason, stack)` instead of `Exception.message/1`. This will show all the information. Often the `stack` is the most useful piece of information.

It would also be good to do `:erlang.raise(kind, reason, stack)` instead of the `{:shutdown, :server_error}` so that link'ed process gets the same exit reason as would occur if the exception was not caught. The `:shutdown` tuple reason could still cause similar issues to the `:transient` problem we discussed when a user accidentally terminates a transient child.

An alternative would be to do the equivalent of
```elixir
catch
  :exit, reason -> {:exit, reason}
  :error, reason -> {:exit, {reason, System.stacktrace()}}
  :throw, reason -> {:exit, {{:nocatch, reason}, System.stacktrace()}}
end
```

Then call exit({exit_reason, {handler_module, :function, args}}). The `Exception` module will format this really nicely with `Exception.format_exit/1` because it takes the form `{exit_reason, {mod, fun, args}}`. The tricky with the `catch` produces the equivalent `exit_reason` to if the process had not caught the exception. We use this heavily in elixir-* projects (especially elixir itself).

I am not sure which is better. An OTP behaviour would usually create the same `exit_reason` (but without the MFA) and exit with that. It would also log a detailed error message that contains the equivalent information to including the mod-fun-args, and maybe some other info too.